### PR TITLE
mgr/dashboard: Unable to login to ceph dashboard until clearing cookies manually

### PR DESCRIPTION
**Pull request by avanthakkar**
This PR fixes the issue with login without manually clearing the cookies. It clears the cookie for token after logout. Also includes some modification in code following SOLID principles of OOPs.

Fixes: https://tracker.ceph.com/issues/49897
Signed-off-by: Avan Thakkar <athakkar@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by avanthakkar** [comment]
jenkins test api

File: src/pybind/mgr/dashboard/controllers/auth.py [hunk 1]
```diff
@@ -89,6 +91,7 @@ def logout(self):
         logger.debug('Logout successful')
         token = JwtManager.get_token_from_header()
         JwtManager.blocklist_token(token)
+        cherrypy.response.cookie['token'] = {'expires': 0, 'max-age': 0}
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/controllers/auth.py:94]: Provided we do the same in  SAML controller, why not just encapsulate this in a protected method
(e.g. ```_delete_auth_token``` or sth similar) in BaseController and call it where appropriate?

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard/controllers/auth.py (hunk 1 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/controllers/auth.py]: Done.

File: src/pybind/mgr/dashboard/controllers/__init__.py [hunk 2]
```diff
@@ -1040,3 +1040,7 @@ def set_cookies(url_prefix, token):
     cherrypy.response.cookie['token']['HttpOnly'] = True
     cherrypy.response.cookie['token']['path'] = '/'
     cherrypy.response.cookie['token']['SameSite'] = 'Strict'
+
+
+def delete_token_cookie():
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/controllers/__init__.py:1045]: What do you think?
```suggestion
def delete_cookie_token():

**Comment by avanthakkar** [comment]
It seems this is the correct approach to expire/delete a cookie https://docs.cherrypy.org/en/3.3.0/progguide/cookies.html#how-to-use-cookies-with-cherrypy. Updated the changes accordingly @alfonsomthd

File: src/pybind/mgr/dashboard/controllers/__init__.py (hunk 2 above)
Comment (avanthakkar) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: Changed it. Thanks @nizamial09

**Comment by avanthakkar** [comment]
jenkins test make check

**Review by nizamial09 — APPROVED** [review]

File: src/pybind/mgr/dashboard/controllers/__init__.py (hunk 2 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: but I still see the name as delete_token_cookie()

Comment (avanthakkar) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: I changed it again as to keep it similar to set_token_cookie and also @alfonsomthd favored  `delete_token_cookie` too. Is that fine with you?

Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/__init__.py]: Ofcourse it is.

**Comment by avanthakkar** [comment]
jenkins test make check

**Review by alfonsomthd — APPROVED** [review]
LGTM Good work!

**Pull request merged**